src/Rector/AbstractRector.php

```diff
@@ -247,6 +247,13 @@ final public function enterNode(Node $node)
         }
 
         if (is_int($refactoredNode)) {
+
+            $this->createdByRuleDecorator->decorate($node, $originalNode, static::class);
```

Comment:
@TomasVotruba this is needed on `is_int($refactoredNode)` usage to be verified to avoid infinte loop. 

On `NodeTraverser::REMOVE_NODE`, probably not needed, or we need to refactor `isNodeRemoved()` usage.